include/vigra/mathutil.hxx

```diff
@@ -164,6 +164,17 @@ VIGRA_DEFINE_MISSING_ABS(signed long long)
 using std::isinf;
 using std::isnan;
 
+#ifdef _LIBCPP_VERSION
+// On older versions of Mac, the system's libc++ does not provide isnan(double)
+// So we just provide it ourselves.
+template <typename T>
+typename std::enable_if<std::is_integral<T>::value, bool>::type
```

Comment:
>And more importantly: why does an integer isnan() solve the problem of missing isnan() for floating types?

So sorry for the confusion.  The comment in the code is incorrect.  The old version of `libc++` DOES provide `isnan(double)`.  It fails to provide `isnan(int)`, etc.  Not sure what I was thinking when I wrote that comment....